packages/datetime/src/dateInput.tsx

```diff
@@ -447,16 +471,49 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
             // the page. tabbing forward should *not* close the popover, because
             // focus will be moving into the popover itself.
             this.setState({ isOpen: false });
+        } else if (e.which === Keys.ESCAPE) {
+            this.setState({ isOpen: false });
+            this.inputRef.blur();
         }
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
+    // blur DOM event listener (not React event)
+    private handlePopoverBlur = () => this.handleClosePopover();
+
+    private registerPopoverBlurHandler = () => {
+        if (this.contentRef != null) {
+            // Popover contents are well structured, but the selector will need
+            // to be updated if more focusable components are added in the future
+            const tabbableElements = this.contentRef.querySelectorAll("input, [tabindex]:not([tabindex='-1'])");
```

Comment:
note to self / @giladgray: we'll need to update this for dom4 v2

Reply:
this is already updated. `queryAll` was removed.